flocx_market/tests/unit/conftest.py

```diff
@@ -33,14 +36,15 @@ def client(app):
 
 @pytest.fixture()
 def db(app):
-    orm.create_all()
-    yield orm
-    orm.drop_all()
+    db_api.setup_db()
+    yield db_api
+    db_api.get_session().query(db_models.Offer).delete()
+    db_api.drop_db()
 
 
 @pytest.fixture()
 def session(db):
-    session = db.session()
+    session = db.get_session()
     yield session
-    if not session.transaction.is_active:
-        session.rollback()
+    # Guess what doesn't work with oslo db sessions...
+    # session.rollback()
```

Comment:
This was necessary because some of the failing tests would result in an invalid transaction, which would cause subsequent tests to fail. Maybe there's another way around it?

Reply:
From what I understand of the oslo db, rollbacks are handled automatically within a session:
https://github.com/malor/oslo.db/blob/master/oslo/db/sqlalchemy/session.py